climada/engine/impact.py

```diff
@@ -487,23 +493,51 @@ def local_exceedance_imp(self, return_periods=(25, 50, 100, 250)):
         if self.imp_mat.size == 0:
             raise ValueError('Attribute imp_mat is empty. Recalculate Impact'
                              'instance with parameter save_mat=True')
+        #check frequency unit
+        if self.frequency_unit in ['1/year', 'annual', '1/y', '1/a']:
+            rp_unit = 'years'
+        elif self.frequency_unit in ['1/month', 'monthly', '1/m']:
+            rp_unit = 'months'
+        elif self.frequency_unit in ['1/week', 'weekly', '1/w']:
+            rp_unit = 'weeks'
+        else:
+            LOGGER.warning(f"Hazard's frequency unit {self.frequency_unit} is not known, "
+                            "years will be used as return period unit.")
+            rp_unit = 'years'
+
         num_cen = self.imp_mat.shape[1]
         imp_stats = np.zeros((len(return_periods), num_cen))
-        cen_step = CONFIG.max_matrix_size.int() // self.imp_mat.shape[0]
-        if not cen_step:
-            raise ValueError('Increase max_matrix_size configuration parameter to > '
-                             f'{self.imp_mat.shape[0]}')
-        # separte in chunks
-        chk = -1
-        for chk in range(int(num_cen / cen_step)):
-            self._loc_return_imp(np.array(return_periods),
-                                 self.imp_mat[:, chk * cen_step:(chk + 1) * cen_step].toarray(),
-                                 imp_stats[:, chk * cen_step:(chk + 1) * cen_step])
-        self._loc_return_imp(np.array(return_periods),
-                             self.imp_mat[:, (chk + 1) * cen_step:].toarray(),
-                             imp_stats[:, (chk + 1) * cen_step:])
-
-        return imp_stats
+
+        for i in range(num_cen):
+            # sort intensties and frequencies at given centroid
+            impact_sorted = np.squeeze(self.imp_mat[:,i].toarray())
+            sorted_idxs = np.argsort(impact_sorted)[::-1]
+            impact_sorted = np.squeeze(impact_sorted[sorted_idxs])
+            frequency_sorted = self.frequency[sorted_idxs]
+            frequency_sorted = np.cumsum(frequency_sorted)
+
+            # fit intensities to cummulative frequencies
+            imp_stats[:,i] = u_fit.calc_fit_interp(
+                1/np.array(return_periods),
+                frequency_sorted,
+                impact_sorted,
+                method=method,
+                x_scale='log',
+                y_thres = 0
+            )
+
+        # create the output GeoDataFrame
+        gdf = gpd.GeoDataFrame(geometry = [Point(x, y) for y, x in self.coord_exp], crs = self.crs)
```

Comment:
I think the following is more efficient and clean:

https://geopandas.org/en/stable/docs/reference/api/geopandas.points_from_xy.html